extensions/azurecore/src/account-provider/azureAccountProvider.ts

```diff
@@ -3,446 +3,155 @@
+		await this.initCompletePromise;
+		const azureAuth = this.getAuthMethod(undefined);
+		return azureAuth?.getSecurityToken(account, resource);
 	}
 
-	private getAccessTokens(account: AzureAccount, resource: azdata.AzureResource): Promise<AzureAccountSecurityTokenCollection> {
-		let self = this;
-
-		const resourceIdMap = new Map<azdata.AzureResource, string>([
-			[azdata.AzureResource.ResourceManagement, self._metadata.settings.armResource.id],
-			[azdata.AzureResource.Sql, self._metadata.settings.sqlResource.id],
-			[azdata.AzureResource.OssRdbms, self._metadata.settings.ossRdbmsResource.id],
-			[azdata.AzureResource.AzureKeyVault, self._metadata.settings.azureKeyVaultResource.id]
-		]);
-
-		let accessTokenPromises: Thenable<void>[] = [];
-		let tokenCollection: AzureAccountSecurityTokenCollection = {};
-		for (let tenant of account.properties.tenants) {
-			let promise = new Promise<void>((resolve, reject) => {
-				let authorityUrl = url.resolve(self._metadata.settings.host, tenant.id);
-				let context = new adal.AuthenticationContext(authorityUrl, null, self._tokenCache);
-
-				context.acquireToken(
-					resourceIdMap.get(resource),
-					tenant.userId,
-					self._metadata.settings.clientId,
-					(error: Error, response: adal.TokenResponse | adal.ErrorResponse) => {
-						// Handle errors first
-						if (error) {
-							// TODO: We'll assume for now that the account is stale, though that might not be accurate
-							account.isStale = true;
-							azdata.accounts.accountUpdated(account);
-
-							reject(error);
-							return;
-						}
-
-						// We know that the response was not an error
-						let tokenResponse = <adal.TokenResponse>response;
+	prompt(): Thenable<azdata.Account | azdata.PromptFailedResult> {
+		return this._prompt();
+	}
 
-						// Generate a token object and add it to the collection
-						tokenCollection[tenant.id] = {
-							expiresOn: tokenResponse.expiresOn,
-							resource: tokenResponse.resource,
-							token: tokenResponse.accessToken,
-							tokenType: tokenResponse.tokenType
-						};
-						resolve();
-					}
-				);
-			});
-			accessTokenPromises.push(promise);
+	private async _prompt(): Promise<azdata.Account | azdata.PromptFailedResult> {
+		await this.initCompletePromise;
+		class Option implements vscode.QuickPickItem {
+			public readonly label: string;
+			constructor(public readonly azureAuth: AzureAuth) {
+				this.label = azureAuth.userFriendlyName;
+			}
 		}
 
-		// Wait until all the tokens have been acquired then return the collection
-		return Promise.all(accessTokenPromises)
-			.then(() => tokenCollection);
-	}
-
-	private getDeviceLoginUserCode(): Thenable<InProgressAutoOAuth> {
-		let self = this;
+		if (this.authMappings.size === 0) {
+			const msg = localize('azure.NoAuthMethod', "No Azure auth method selected");
+			console.log('noAuthMethodSelected');
```

Comment:
Do we really need to log this?

Reply:
It's just so when we get bug reports in the future, more logs isn't a problem potentially?

Reply:
I guess - should probably consider logging to a separate stream though if we care about that so we don't have to pick through console logging. Anyways fine with it either way - although I'd prefer you make the output more descriptive since someone seeing this in the logs would have absolutely no idea what this is referring to unless they knew the code. 

---
In reply to: [395911374](https://github.com/microsoft/azuredatastudio/pull/9664#discussion_r395911374) [](ancestors = 395911374)

Reply:
That makes sense, i'll update the message.

If it helps this is a serious edgecase if you have multiple auth methods enabled. (Default will just be one)